libraries/FunctionalInterrupt/examples/Functional/Functional.ino

```diff
@@ -0,0 +1,88 @@
+#include <FunctionalInterrupt.h>
+
+#if defined(ESP32)
+#define BUTTON1 16
+#define BUTTON2 17
+#elif defined(ARDUINO_ESP8266_WEMOS_D1MINI)
+#define BUTTON1 D4
+#define BUTTON2 D3
+#else
+#define BUTTON1 2
+#define BUTTON2 0
+#endif
+
+class Button {
+  public:
+    Button(uint8_t reqPin) : PIN(reqPin) {
+      pinMode(PIN, INPUT_PULLUP);
+      // Arduino API:
+      //attachInterruptArg(PIN, [](void* self) {
+      //  static_cast<Button*>(self)->buttonIsr();
+      //}, this, FALLING); // fails on ESP8266: "buttonIsr not in IRAM"
+      //attachInterruptArg(PIN, reinterpret_cast<void(*)(void*)>(&buttonIsr_static), this, FALLING); // works on ESP8266
+      attachInterrupt(PIN, std::bind(&Button::buttonIsr, this), FALLING); // works on ESP8266
+      // FunctionalInterrupt.h API:
+      //attachScheduledInterrupt(PIN, [this](const InterruptInfo & ii) {
```

Comment:
What about a second example, this one called `InterruptFunctional.ino` and the commented one `ScheduledInterruptFunctional.ino` ?